### PR TITLE
ci: enable sentry in the release builds

### PR DESCRIPTION
**Pull request by markoburcul**
Referenced issue: https://github.com/status-im/status-desktop/issues/16842

### What does the PR do

Adds Sentry related secrets to env vars for release builds.

### Affected areas

Desktop release build.

### Architecture compliance

- [ ] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

### CODE REVIEW

File: ci/Jenkinsfile.combined [hunk 1]
```diff
@@ -32,6 +32,13 @@ pipeline {
 
   stages {
     stage('Build') {
+      environment {
+        SENTRY_PRODUCTION = utils.isReleaseBuild() ? true : false
+      }
+      withCredentials([
+        string(credentialsId: 'status-desktop-sentry-dsn', variable: 'STATUS_BUILD_SENTRY_DSN_STATUS_DESKTOP'),
+        string(credentialsId: 'status-go-sentry-dsn', variable: 'STATUS_BUILD_SENTRY_DSN_STATUS_GO')
+      ])
```
Comment (jakubgs) [inline ci/Jenkinsfile.combined:41]: Why are you adding this here? It's already done in `status-jenkins-lib`.

**Review by igor-sirotin — APPROVED** [review]
Nice, thank you!
I suppose `utils.isReleaseBuild()` is defined externally and distringuishes `ci.infra.status.im` from `ci.status.im`, right?

**Comment by markoburcul** [comment]
> I suppose `utils.isReleaseBuild()` is defined externally and distringuishes `ci.infra.status.im` from `ci.status.im`, right?

Yes, it is defined in the jenkins library and returns true if one of the following conditions is met:
```
    params.RELEASE == true ||
    params.BUILD_TYPE == 'release' ||
    env.JOB_NAME.startsWith('status-mobile/release') ||
    env.JOB_NAME.startsWith('status-desktop/release') ||
    env.JOB_NAME.startsWith('status-node/release')
```

**Comment by igor-sirotin** [comment]
Tested MacOS build from here: https://ci.infra.status.im/job/status-desktop/job/release/job/release%2Fsentry-test-panic-button/1/
Which has the `Trigger panic` button from here: https://github.com/status-im/status-desktop/pull/16850

Sentry report received ✅ 
image

But the environment is not set to `production` 🤔 
Will double check where's the error with this.

**Comment by igor-sirotin** [comment]
@markoburcul I've fixed the issue and updated the branch.
Can you please start another release build from https://github.com/status-im/status-desktop/pull/16850? 

fix here: https://github.com/status-im/status-go/pull/6149

File: ci/Jenkinsfile.combined [hunk 2]
```diff
@@ -32,6 +32,9 @@ pipeline {
 
   stages {
     stage('Build') {
+      environment {
+        SENTRY_PRODUCTION = "${utils.isReleaseBuild() ? 'true' : 'false'}"
```
Comment (jakubgs) [inline ci/Jenkinsfile.combined:36]: Why is this in the combined `Jenkinsfile`? It will not affect sub-jobs spawned by this job.

**Review by jakubgs — CHANGES_REQUESTED** [review]

**Comment by markoburcul** [comment]
> @markoburcul I've fixed the issue and updated the branch. Can you please start another release build from #16850?
> 
> fix here: [status-im/status-go#6149](https://github.com/status-im/status-go/pull/6149)

The build with newest updates: https://ci.infra.status.im/job/status-desktop/job/release/job/release%252Fsentry-test-panic-button/2/

**Comment by igor-sirotin** [comment]
> The build with newest updates: https://ci.infra.status.im/job/status-desktop/job/release/job/release%252Fsentry-test-panic-button/2/

Thank you @markoburcul! I've tested the new build.

## TLDR

PR is good to go, thank you 🎉 
But would be nice to set up Github Integration in Sentry.

## Fixed things

1. `production` environment is now there ✅ 

    image

2. Stacktrace is properly there ✅ 
    
    image
    
## Minor issue

I noted that the stacktrace doesn't contain code snippets at all 🤔 

For local crash reports, stacktrace looks like this:

image

Because each stacktrace frame contains a code snippet in the report:
    
image

But it's not the case for release builds.

I believe it's **better**, because it uses less network data for users.
Though it's handy to have, I wonder if Sentry Github integration will fix this on our side.

@markoburcul can you please enable it? At least for the `status-go` repository.

image

**Comment by markoburcul** [comment]
> @markoburcul can you please enable it? At least for the `status-go` repository.

I get 404 page not found on GitHub when I try to add integration. Maybe its because I lack rights on our organization/enterprise. Maybe @jakubgs can try it?

**Comment by jakubgs** [comment]
It won't work just by itself. For an integration with GitHub to work our Sentry instance needs to be registered as a GitHub App:
https://develop.sentry.dev/integrations/github/

File: ci/Jenkinsfile.linux [hunk 3]
```diff
@@ -73,6 +73,7 @@ pipeline {
     STATUS_CLIENT_TARBALL = "pkg/${utils.pkgFilename(ext: 'tar.gz', arch: getArch())}"
     /* prevent sharing cache dir across different jobs */
     GO_GENERATE_FAST_DIR = "${env.WORKSPACE_TMP}/go-generate-fast"
+    SENTRY_PRODUCTION = "${utils.isReleaseBuild() ? 'true' : 'false'}"
```
Comment (jakubgs) [inline ci/Jenkinsfile.linux:76]: How is this env variable actually checked? Are we sure a `false` string won't just be treated as true?

Comment (igor-sirotin) [reply ci/Jenkinsfile.linux:76]: It's checked for `true` and `1`:
https://github.com/status-im/status-go/blob/987a9e8707e97c1757f87f621c036d671bb046fe/internal/sentry/params.go#L34

**Review by jakubgs — APPROVED** [review]

**Comment by markoburcul** [comment]
> It won't work just by itself. For an integration with GitHub to work our Sentry instance needs to be registered as a GitHub App:
> https://develop.sentry.dev/integrations/github/

I don't have enough permissions on Status org to add the Github App for Sentry..

**Comment by igor-sirotin** [comment]
@markoburcul @jakubgs I opened a separate issue to track this: 
- https://github.com/status-im/status-go/issues/6152
